feat: enable resizing the sidebar (#1078)
puckeditor/puck: apps/docs/pages/docs/api-reference/data-model/app-state.mdx

---

### `ui.rightSideBarVisible`

Whether or not the right side bar is visible.

---

### `ui.leftSideBarWidth`

> Width of the left side bar in pixels. This value can be dynamically adjusted by users through the `ResizeHandle` component, which provides a draggable interface for resizing the sidebar.

Reviewer: ```suggestion
Current width of the left side bar in pixels.
```

ResizeHandle is not part of the public API.

---

### `ui.rightSideBarWidth`

> Width of the right side bar in pixels. Similar to the left sidebar, this width can be adjusted using the `ResizeHandle` component.

Reviewer: ```suggestion
Current width of the right side bar in pixels.
```

ResizeHandle is not part of the public API.

> 
> ---
> 
> ### `ResizeHandle`
> 
> The `ResizeHandle` component provides a draggable interface for resizing sidebars in the Puck editor. It appears as a thin vertical handle on the edge of sidebars that users can drag to adjust the width.
> 
> | Prop          | Type                                  | Description                                                  |
> | ------------- | ------------------------------------- | ------------------------------------------------------------ |
> | `position`    | `"left" \| "right"`                   | Determines which side of the sidebar the handle appears on   |
> | `sidebarRef`  | `{ current: HTMLDivElement \| null }` | React ref to the sidebar element being resized               |
> | `onResize`    | `(width: number) => void`             | Callback function triggered during resize with the new width |
> | `onResizeEnd` | `(width: number) => void`             | Callback function triggered when resizing ends               |
> 
> The component creates a draggable handle that updates the corresponding sidebar width in the AppState when dragged, and resets the canvas zoom when the resize operation completes.

Reviewer: This is not part of the public API.

```suggestion
```

> ---

Reviewer: This is not public so it's not necessary to document. We only need `rightSideBarWidth` and `leftSideBarWidth` added

Author: removed ✅

---

### `ui.rightSideBarVisible`

Whether or not the right side bar is visible.

> ---

Reviewer: I understand you reordered these to keep them group, but these docs are alphabetical.

If you would rather not worry about docs, feel free to drop these changes and we can handle in post.
